editor/debugger/script_editor_debugger.cpp

```diff
@@ -470,6 +470,10 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 	} else if (p_msg == "error") {
 		DebuggerMarshalls::OutputError oe;
 		ERR_FAIL_COND_MSG(oe.deserialize(p_data) == false, "Failed to deserialize error message");
+		if (oe.type == DebuggerMarshalls::ERROR_TYPE_GDSCRIPT_WARNING && !ProjectSettings::get_singleton()->get_setting("debug/gdscript/warnings/enable")) {
```

Comment:
This would be better to handle further up the call stack in the GDScript module.
The debugger shouldn't need to know about and special-case GDScript.

Reply:
OK, then this need to be completely rewritten...